tests/config-override.test.sh

```
@@ -631,6 +693,113 @@ case $rc in
+EOF
+  out="$(uberdev_read_int_in_range review.confidence_threshold UBERDEV_REVIEW_THRESHOLD 0 100 80)"
+  echo "U10_RESULT=[$out]" >&2
+  echo "U10_AUDIT=[$(cat .uberdev/audit.jsonl)]" >&2
+'
+_u10_expect 80 "U10.4: non-integer abc -> default 80"
+grep -qE "review.confidence_threshold = .abc. is invalid \(non_integer\)" <<<"$_LAST_STDERR" \
+  && pass "U10.4: stderr reason is non_integer" \
+  || fail "U10.4: non_integer reason missing"
+grep -qF '"event":"uberdev_config_invalid"' <<<"$_LAST_STDERR" \
+  && grep -qF '"key":"review.confidence_threshold"' <<<"$_LAST_STDERR" \
+  && pass "U10.4: uberdev_config_invalid audit row carries the key" \
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)